pkg/tests/tasks/traffic/circuit_breaking_test.go

```diff
@@ -39,76 +46,95 @@ func cleanupCircuitBreaking() {
 }
 
 func TestCircuitBreaking(t *testing.T) {
-	test.NewTest(t).Id("T6").Groups(test.Full, test.InterOp).NotRefactoredYet()
+	NewTest(t).Id("T6").Groups(Full, InterOp).Run(func(t TestHelper) {
+		hack.DisableLogrusForThisTest(t)
+		ns := "bookinfo"
+		t.Cleanup(func() {
+			oc.RecreateNamespace(t, ns)
+		})
+
+		app.InstallAndWaitReady(t, app.Httpbin(ns), app.Fortio(ns))
+
+		t.NewSubTest("TrafficManagement_tripping").Run(func(t TestHelper) {
+			t.Log("verify traffic management tripping circuit breaker")
+			t.LogStep("Configure circuit breaker destination rule")
+			oc.ApplyString(t, ns, httpbinCircuitBreaker)
+
+			t.LogStep("Verify connection with curl: expected 200 OK")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.Exec(t,
+					pod.MatchingSelector("app=fortio", ns),
+					"fortio",
+					"/usr/bin/fortio curl -quiet http://httpbin:8000/get",
+					assert.OutputContains("200",
+						"Got expected response from httpbin: 200 OK",
+						"ERROR: Got unexpected response from httpbin not 200 OK"))
+			})
+
+			t.LogStep("Tripping the circuit breaker")
+			t.Log("To tipping the circuit breaker we are going to send 50 requests to httpbin with 2 connections")
+			connection := 2
+			reqCount := 50
+			msg := oc.Exec(t,
+				pod.MatchingSelector("app=fortio", ns),
+				"fortio",
+				fmt.Sprintf("/usr/bin/fortio load -c %d -qps 0 -n %d -loglevel Warning http://httpbin:8000/get", connection, reqCount))
+
+			t.LogStep("Validate the number of 200 responses")
+			t.Log("verify from output message the number of 200 responses to the load test. We expect to have a line with this information: Code 200 : XX (X0.0 %)")
+			c200 := getNumberOfResponses(t, msg, `Code 200.*`)
+
+			t.LogStep("Validate the number of 503 responses")
+			t.Log("verify from output message the number of 500 responses to the load test. We expect to have a line with this information: Code 503 : XX (X0.0 %)")
+			c503 := getNumberOfResponses(t, msg, `Code 503.*`)
+
+			t.LogStep("Validate the percentage of 200 responses and 503 to the total of requests")
+			t.Log("We expect to have 60% 200 responses and 40% 503 responses")
+			tolerance := 0.5
+			if util.IsWithinPercentage(c200, reqCount, 0.6, tolerance) && util.IsWithinPercentage(c503, reqCount, 0.4, tolerance) {
+				t.Logf(
+					"Success. Circuit breaking acts as expected. "+
+						"Code 200 hit %d of %d, Code 503 hit %d of %d", c200, reqCount, c503, reqCount)
```

Comment:
Considering that the previous log statement says `We expect to have 60% 200 responses...`, it would be nice if this also printed the percentage, so you don't have to calculate it in your head.

Reply:
Done